app.rb

```diff
@@ -1,99 +1,91 @@
 require_relative 'book'
 require_relative 'person'
 require_relative 'teacher'
-require_relative 'rental'
 require_relative 'student'
+require_relative 'rental'
 
 class App
-  attr_accessor :all_books, :all_people, :all_rentals
-
   def initialize
-    @all_books = []
-    @all_people = []
-    @all_rentals = []
+    @persons = []
+    @books = []
+    @rentals = []
   end
 
-  def books
-    @all_books.each_with_index do |book, index|
-      puts "#{index}) #{book}"
+  # create person
+  def create_person(user_type)
+    print 'Age:'
+    age = gets.chomp.to_i
+    print 'Name:'
+    name = gets.chomp
+
+    case user_type
+    when '1'
+      create_student(age, name)
+    when '2'
+      create_teacher(age, name)
+    else
+      puts 'Invalid choice'
     end
   end
 
-  def people
-    @all_people.each_with_index do |person, index|
-      puts "#{index}) #{person}"
-    end
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    permission = %w[Y y].include?(permission)
+
+    student = Student.new(age, name, parent_permission: permission)
+    @persons << student
+    puts 'Student created successfully'
   end
```

Comment:
- [OPTIONAL] Great job working so far and the functionality is working fine ✨. but I think it would be great if you just kept the functionality of create student and create teacher since this is considered a Bad example of a Single responsibility rule and a Bad example of YAGNI and KISS as the contents of this method could've been placed somewhere else so kindly read more about [Best practices and Solid lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)  🔥💪🏻✨🤩

- You can check how the [refactor](https://github.com/Reem-lab/School-library-ruby/pull/5/files#diff-df46c71dda0c06fcf280fa5a3c58b7f74f08c2c83d8c4dd07632310ae9b700a7) is 


- [ ] I think you could refactor the functionality inside these methods like the inputs to make it better and not redundant ✨🙏🏻